suite-common/wallet-core/src/blockchain/blockchainMiddleware.ts

```diff
@@ -0,0 +1,55 @@
+import { getUnixTime } from 'date-fns';
+
+import { createMiddlewareWithExtraDeps } from '@suite-common/redux-utils';
+import { BLOCKCHAIN as TREZOR_CONNECT_BLOCKCHAIN_ACTIONS, BlockchainEvent } from '@trezor/connect';
+
+import {
+    onBlockchainConnectThunk,
+    onBlockMinedThunk,
+    onNotificationThunk,
+    updateFeeInfoThunk,
+} from './blockchainThunks';
+
+export const blockchainMiddleware = createMiddlewareWithExtraDeps(
+    (action: BlockchainEvent, { dispatch, next, extra }) => {
+        // propagate action to reducers
+        next(action);
+
+        const { validatePendingTxOnBlock, fetchTrezorPools } = extra.thunks;
+
+        switch (action.type) {
+            case TREZOR_CONNECT_BLOCKCHAIN_ACTIONS.CONNECT:
+                dispatch(onBlockchainConnectThunk(action.payload.coin.shortcut));
+
+                // once suite connects to blockchain, fetch additional data required
+                // for cardano staking if applicable
+                if (['ADA', 'tADA'].includes(action.payload.coin.shortcut)) {
+                    dispatch(fetchTrezorPools(action.payload.coin.shortcut as 'ADA' | 'tADA'));
+                }
+                break;
+            case TREZOR_CONNECT_BLOCKCHAIN_ACTIONS.BLOCK:
+                dispatch(updateFeeInfoThunk(action.payload.coin.shortcut));
+                dispatch(onBlockMinedThunk(action.payload));
+                // cardano stuff
+                dispatch(
+                    validatePendingTxOnBlock({
+                        block: action.payload,
+                        ts: getUnixTime(new Date()),
+                    }),
+                );
+                break;
+            case TREZOR_CONNECT_BLOCKCHAIN_ACTIONS.NOTIFICATION:
+                dispatch(onNotificationThunk(action.payload));
+                break;
+            case TREZOR_CONNECT_BLOCKCHAIN_ACTIONS.ERROR:
+                // TODO BEFORE MERGE: figure out if this should be here, it throws TS error and from implementation
```

Comment:
I have found the comment:

> // call TrezorConnect.unsubscribe, it doesn't cost anything and should emit BLOCKCHAIN.CONNECT or BLOCKCHAIN.ERROR event

BLOCKCHAIN.ERROR is TREZOR_CONNECT_BLOCKCHAIN_ACTIONS.ERROR: in your case so maybe try to investigate it more.